js/src/visitor/get.ts

```diff
@@ -265,9 +267,11 @@ const getDictionary = <T extends Dictionary>(data: Data<T>, index: number): T['T
 /* istanbul ignore next */
 /** @ignore */
 const getInterval = <T extends Interval>(data: Data<T>, index: number): T['TValue'] =>
-    (data.type.unit === IntervalUnit.DAY_TIME)
-        ? getIntervalDayTime(data as Data<IntervalDayTime>, index)
-        : getIntervalYearMonth(data as Data<IntervalYearMonth>, index);
+    (data.type.unit === IntervalUnit.MONTH_DAY_NANO)
+        ? getIntervalMonthDayNano(data as Data<IntervalMonthDayNano>, index)
+        : (data.type.unit === IntervalUnit.DAY_TIME)
+            ? getIntervalDayTime(data as Data<IntervalDayTime>, index)
+            : getIntervalYearMonth(data as Data<IntervalYearMonth>, index);
```

Comment:
Could we split these conditions out into separate methods and assign them to the GetVisitor prototype separately?